controllers/provider-aws/pkg/webhook/controlplaneshoot/validator.go

```diff
@@ -0,0 +1,155 @@
+}
+
+func validateKubeAPIServerDeployment(dep *appsv1.Deployment) error {
+	ps := &dep.Spec.Template.Spec
+	if c := controlplane.ContainerWithName(ps.Containers, "kube-apiserver"); c != nil {
+		ensureKubeAPIServerCommandLineArgs(c)
+		ensureEnvVars(c)
+		ensureVolumeMounts(c)
+	}
+	ensureVolumes(ps)
+	return nil
+}
+
+func validateKubeControllerManagerDeployment(dep *appsv1.Deployment) error {
+	ps := &dep.Spec.Template.Spec
+	if c := controlplane.ContainerWithName(ps.Containers, "kube-controller-manager"); c != nil {
+		ensureKubeControllerManagerCommandLineArgs(c)
+		ensureEnvVars(c)
+		ensureVolumeMounts(c)
+	}
+	ensureVolumes(ps)
+	return nil
+}
+
+func ensureKubeAPIServerCommandLineArgs(c *corev1.Container) {
+	c.Command = controlplane.EnsureStringWithPrefix(c.Command, "--cloud-provider=", "aws")
+	c.Command = controlplane.EnsureStringWithPrefix(c.Command, "--cloud-config=",
+		"/etc/kubernetes/cloudprovider/cloudprovider.conf")
+	c.Command = controlplane.EnsureStringWithPrefixContains(c.Command, "--enable-admission-plugins=",
+		"PersistentVolumeLabel", ",")
+	c.Command = controlplane.EnsureNoStringWithPrefixContains(c.Command, "--disable-admission-plugins=",
+		"PersistentVolumeLabel", ",")
+}
+
+func ensureKubeControllerManagerCommandLineArgs(c *corev1.Container) {
+	c.Command = controlplane.EnsureStringWithPrefix(c.Command, "--cloud-provider=", "external")
+	c.Command = controlplane.EnsureStringWithPrefix(c.Command, "--cloud-config=",
+		"/etc/kubernetes/cloudprovider/cloudprovider.conf")
+	c.Command = controlplane.EnsureStringWithPrefix(c.Command, "--external-cloud-volume-plugin=", "aws")
+}
+
+func ensureEnvVars(c *corev1.Container) {
+	c.Env = controlplane.EnsureEnvVarWithName(c.Env,
+		corev1.EnvVar{
+			Name: "AWS_ACCESS_KEY_ID",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key:                  aws.AccessKeyID,
+					LocalObjectReference: corev1.LocalObjectReference{Name: common.CloudProviderSecretName},
+				},
+			},
+		},
+	)
+	c.Env = controlplane.EnsureEnvVarWithName(c.Env,
+		corev1.EnvVar{
+			Name: "AWS_SECRET_ACCESS_KEY",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key:                  aws.SecretAccessKey,
+					LocalObjectReference: corev1.LocalObjectReference{Name: common.CloudProviderSecretName},
+				},
+			},
+		},
+	)
+}
+
+func ensureVolumeMounts(c *corev1.Container) {
+	c.VolumeMounts = controlplane.EnsureVolumeMountWithName(c.VolumeMounts,
+		corev1.VolumeMount{
+			Name:      common.CloudProviderSecretName,
+			MountPath: "/etc/kubernetes/cloudprovider",
+		},
+	)
+	c.VolumeMounts = controlplane.EnsureVolumeMountWithName(c.VolumeMounts,
+		corev1.VolumeMount{
+			Name:      common.CloudProviderSecretName,
+			MountPath: "/srv/cloudprovider",
+		},
+	)
+}
+
+func ensureVolumes(ps *corev1.PodSpec) {
+	ps.Volumes = controlplane.EnsureVolumeWithName(ps.Volumes,
+		corev1.Volume{
+			Name: common.CloudProviderConfigName,
```

Comment:
Added it to `controllers/provider-aws/pkg/aws`, I think it's the best place since it already contains similar constants. The new constant is used by both `webhook/controlplane` and `controller/controlplane`.